### PR TITLE
[FEATURE] Build Badges

### PR DESCRIPTION
**Pull request by MatejMecka**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

I've made a web page that gets the latest commit and displays the status based on it.

### CODE REVIEW

File: mod_test/controllers.py [hunk 1]
```diff
@@ -283,3 +284,24 @@ def download_build_log_file(test_id):
...
+
+
+# Define a web page to be accessed
+@mod_test.route('/svg/latest/')
+def svgbadge():
+    dwg = svgwrite.Drawing('status.svg', profile='full', size=('66px','18px')) #  dimensions from circle ci
+    # Get the latest Test
+    commit = Test.query.filter(Test.test_type != TestType.pull_request).first()
+    # Check it's status
+    if commit.TestStatus == TestStatus.completed:
```
Comment (canihavesomecoffee) [inline mod_test/controllers.py]: It's not because a test is completed that it passed.

File: mod_test/controllers.py [hunk 2]
```diff
@@ -283,3 +284,24 @@ def download_build_log_file(test_id):
...
+        dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
+        dwg.save()
+    elif commit.TestStatus == TestStatus.canceled:
+        dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
+        dwg.save()
+    else:
+        dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
+        dwg.save()
+    response = make_response(open("status.svg").read())
```
Comment (canihavesomecoffee) [inline mod_test/controllers.py]: So you write an SVG to the disk, read it out again EACH time a request to the page gets made? That's horribly inefficient.

File: mod_test/controllers.py [hunk 3]
```diff
@@ -283,3 +284,24 @@ def download_build_log_file(test_id):
         raise TestNotFoundException('Build log for Test {id} not found'.format(id=test_id))
 
     raise TestNotFoundException('Test with id {id} not found'.format(id=test_id))
+
+
+# Define a web page to be accessed
+@mod_test.route('/svg/latest/')
+def svgbadge():
+    dwg = svgwrite.Drawing('status.svg', profile='full', size=('66px','18px')) #  dimensions from circle ci
```
Comment (canihavesomecoffee) [inline mod_test/controllers.py]: Where did you get these dimensions from?

File: mod_ci/controllers.py [hunk 4]
```diff
@@ -777,12 +778,20 @@ def progress_reporter(test_id, token):
...
                         state = Status.SUCCESS
                         message = 'Tests completed'
-
+                        if test.test_type != test.pull_request:
+                            dwg = svgwrite.Drawing()
+                            dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
+                            dwg.save()
                 else:
-                    message = progress.message
+                    message = progress.message:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Why is this added? This will break the code.

**Comment by saurabhshri** [comment]
I just commented on the issue : https://github.com/CCExtractor/sample-platform/issues/111#issuecomment-355594495

File: mod_ci/controllers.py [hunk 5]
```diff
@@ -620,6 +621,18 @@ def start_ci():
         return json.dumps({'msg': 'EOL'})
 
 
+def check_status_for_badge(status):
+    if test.test_type != test.pull_request:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This won't work.

File: mod_ci/controllers.py [hunk 6]
```diff
@@ -620,6 +621,18 @@ def start_ci():
...
+        dwg = svgwrite.Drawing('status.svg', profile='full')
+        if status == Status.SUCCESS:
+            dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
+        elif status == Status.FAILURE:
+            dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
+        else:
+            dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
+        dwg.save()  # Couldn't give it an external path it saved it in the local directory...
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: That's most likely because you didn't define the path when you created the SVG file ;)

File: mod_ci/controllers.py [hunk 7]
```diff
@@ -620,6 +621,18 @@ def start_ci():
...
+        shutil.move('status.svg', '../static/status.svg')
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This shouldn't be necessary (see comment above).

File: mod_ci/controllers.py [hunk 8]
```diff
@@ -620,6 +621,18 @@ def start_ci():
         return json.dumps({'msg': 'EOL'})
 
 
+def check_status_for_badge(status):
+    if test.test_type != test.pull_request:
+        dwg = svgwrite.Drawing('status.svg', profile='full')
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: You don't take into account the platform.

File: static/svg/readme.txt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+the only purpose of this file is for the svg folder to exist.
```
Comment (canihavesomecoffee) [inline static/svg/readme.txt]: I'd prefer an empty `.keep` file, which is an unofficial standard.

File: mod_ci/controllers.py (hunk 6 above)
Comment (MatejMecka) [reply mod_ci/controllers.py]: when i tested it i gave it external paths and always generated where the script was executed

Comment (canihavesomecoffee) [reply mod_ci/controllers.py]: According to the docs, it's possible: http://pythonhosted.org/svgwrite/classes/drawing.html

File: mod_ci/controllers.py [hunk 10]
```diff
@@ -777,10 +789,12 @@ def progress_reporter(test_id, token):
                     if crashes > 0 or results > 0:
                         state = Status.FAILURE
                         message = 'Not all tests completed successfully, please check'
+                        check_status_for_badge(state, test)
+
                     else:
                         state = Status.SUCCESS
                         message = 'Tests completed'
-
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Please add the empty line again after the `check_status...` line.

File: mod_ci/controllers.py [hunk 11]
```diff
@@ -620,6 +621,19 @@ def start_ci():
         return json.dumps({'msg': 'EOL'})
 
 
+def check_status_for_badge(status, test):
+    if test.test_type != test.pull_request:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This won't work You're comparing against the wrong type.
